File: presentation/src/main/java/io/tripled/social/client/presentation/controller/ReadController.java
package io.tripled.social.client.presentation.controller;

import io.tripled.social.client.application.ReadMessagesUseCase;
import io.tripled.social.client.presentation.Controller;
import io.tripled.social.client.presentation.Output;
import io.tripled.social.client.presentation.request.ReadCliRequest;

public class ReadController implements Controller<ReadCliRequest> {

  private ReadMessagesUseCase readMessagesApplicationService;

  public ReadController(ReadMessagesUseCase service) {
    this.readMessagesApplicationService = service;
  }

  @Override
  public void execute(ReadCliRequest request, Output output) {
    output.print(readMessagesApplicationService.readMessage(request.getUserName()));
  }
}

File: src/main/kotlin/io/tripled/social/client/infrastructure/cli/controller/ReadController.kt
package io.tripled.social.client.infrastructure.cli.controller

import io.tripled.social.client.application.ReadMessages
import io.tripled.social.client.infrastructure.cli.Controller
import io.tripled.social.client.infrastructure.cli.Output
import io.tripled.social.client.infrastructure.cli.request.ReadCliRequest

class ReadController(private val readMessagesApplicationService: ReadMessages) : Controller<ReadCliRequest> {
    override fun execute(request: ReadCliRequest, output: Output) {
        output.print(readMessagesApplicationService.readMessage(request.userName))
    }
}
